search_service/proxy/elasticsearch.py

```diff
@@ -377,6 +376,130 @@ def fetch_table_search_results(self, *,
                                    query_name=query_name,
                                    model=Table)
 
+    @staticmethod
+    def convert_query_json_to_query_dsl(search_request: dict,
+                                        query_term: str) -> str:
+        """
+        Convert the generic query json to query DSL
+        e.g
+        ```
+        {
+            'type': 'AND'
+            'filters': {
+                'database': ['hive', 'bigquery'],
+                'schema': ['test-schema1', 'test-schema2'],
+                'table': ['*amundsen*'],
+                'column': ['*ds*']
+                'tag': ['test-tag']
+            }
+        }
+
+        This generic JSON will convert into DSL depending on the backend engines.
+
+        E.g in Elasticsearch, it will become
+        'database':('hive' OR 'bigquery') AND
+        'schema':('test-schema1' OR 'test-schema2') AND
+        'table':('*amundsen*') AND
+        'column':('*ds*') AND
+        'tag':('test-tag')
+        ```
+
+        :param search_request:
+        :return: The search engine query DSL
+        """
+        filter_list = search_request.get('filters')
```

Comment:
this is a general nit for this code as a whole, but it would be awesome to see more typing hints! Especially in places like this, where its not immediately clear what type 'filters' will be

Reply:
true, doesn't help that this actually a `dict` and not a list. I will add typing hints for any new code that uses the `.get()` logic

Reply:
Yes please! Type hint all the things =)